File: EOCV-Sim/src/main/java/com/github/serivesmejia/eocvsim/util/fps/FpsCounter.java
package com.github.serivesmejia.eocvsim.util.fps;

import com.qualcomm.robotcore.util.ElapsedTime;

public class FpsCounter {

    private ElapsedTime elapsedTime = new ElapsedTime();

    private volatile int fpsC;
    private volatile int lastFps;

    public synchronized void update() {
        fpsC++;
        if(elapsedTime.seconds() >= 1) {
            lastFps = fpsC; fpsC = 0;
            elapsedTime.reset();
        }
    }

    public synchronized int getFPS(){
        return lastFps;
    }

}

File: EOCV-Sim/src/main/java/com/github/serivesmejia/eocvsim/util/fps/FpsCounter.kt
package com.github.serivesmejia.eocvsim.util.fps

import com.qualcomm.robotcore.util.ElapsedTime

class FpsCounter {

    private val elapsedTime = ElapsedTime()

    @Volatile
    private var fpsC = 0

    @get:Synchronized
    @Volatile
    var fps = 0
        private set

    @Synchronized
    fun update() {
        fpsC++
        if (elapsedTime.seconds() >= 1) {
            fps = fpsC
            fpsC = 0
            elapsedTime.reset()
        }
    }

}